### DISCUSSION TITLE
Allow {{ table }} variable in a model stub + document studs

### DISCUSSION THREAD

**Discussion by realtebo** [laravel/framework#48674] [Ideas]
I am porting to laravel a huge project where all models need to customize table and id

i would like to could make a stub like this

```
<?php

namespace {{ namespace }};

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class {{ class }} extends Model
{
    use HasFactory;

    protected $table = "{{ table }}";

    protected $primaryKey = "id_{{ table}}";

    /*************************************
     *              relations
     ***********************************/
}

```

But `make:model` command is not populating the `table` placeholder

Also, I cannot find any doc about which placeholders are available in which stub

**Comment by shaedrich** [comment]
Although nice to have in your use case, the problem with stubs is, they use simple string replacement and therefore, when a placeholder should have a default value, it cannot simply be omitted, which would all models to have redundant properties when one hasn't your use case.

**Reply by realtebo** [reply]
I do not want all stub to have it.
I would like to have option to create a stub with {{ table }} placeholder.
So it's retrocompatible and without adding useless thing to actual stubs.
I can create new stubs, but the code who generates real model classes from stub is not, now, able to use {{ table }} placeholder

**Comment by mohamedhekal** [comment]
You're right, the default model stub that ships with Laravel doesn't have a {{table}} placeholder. To add custom placeholders like that, you'll need to publish and modify the stub yourself.

Here are the steps:

Publish the model stub:
php artisan vendor:publish --tag=laravel-stubs

1. Open the stub in resources/stubs/model.stub
2. Add your custom placeholders like {{table}} where needed.
3. Update the model factory in database/factories/ModelFactory.php to pass the custom values:

$factory->define(App\Models\{{class}}, function () {
    return [
        // ...
        'table' => '{{table}}',
    ];
});

4. Generate your model like normal:

php artisan make:model MyModel -f

The -f flag tells it to use the factory, which will populate the custom placeholders.

**Comment by realtebo** [comment]
thank you, but what is `database/factories/ModelFactory.php` ? 

Is there something to publish?
